src/app/clusters/access-control-server/access-control-server.cpp

```diff
@@ -473,6 +601,16 @@ CHIP_ERROR AccessControlAttribute::Write(const ConcreteDataAttributePath & aPath
     return ChipErrorToImErrorMap(WriteImpl(aPath, aDecoder));
 }
 
+#if CHIP_CONFIG_USE_ACCESS_RESTRICTIONS
+void OnPlatformEventHandler(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    if (event->Type == DeviceLayer::DeviceEventType::kCommissioningComplete)
```

Comment:
This happens on CommissioningComplete command, which ends a fail-safe, which can occur outside of "commissioning flows" (as in joing a new fabric) and can arise when reconfiguring networking or doing an UpdateNOC, both of which require ArmFailSafe/CommissioningComplete pairs. 

I don't think there is any need to be stateful on "commissioning flow".

Reply:
The spec actually does say that review fabric restrictions is disallowed during commissioning. If the device adds ARL attributes before the fabric is added then it has to be added to the failsafe context in order to be properly rolled back. When I last talked to Chris, he said he was opting to disallow ARL changes during commissioning rather than play with the failsafe context and therefore the CommissioningARL restrictions would continue to be in effect until the commissioning complete was received IF the accessing fabric matches the fabric in the failsafe context.

The switch over from CommissioningARL to ARL I think IS required to happen at commissioning complete if the client is on the new fabric. Because otherwise there's a window of time where restrictions don't apply (ie, you're on a new fabric, but new fabric ARL entires have not yet been added). If you're swapping over to ARL directly after AddNOC, then the ARL entries need to be part of the failsafe. 

Incidentally, the description of CommissioningARL attribute in the text contradicts the description of the CommissioningARL in the ARL attribute. So that needs clarification as to exactly when this switch over applies.

Reply:
Which brings up another question - are the ARL entries for a fabric properly cleared when a fabric is removed, and are there tests for that?

Reply:
The ARL entries are not "stored" persistently (or shouldn't be, in real products). They should reflect a current state.

Regarding the race window, I think it's fair that the CommissioningARL applies:

- Whenever the session is a PASE session without an accessing fabric
- Whenever the session is a CASE session for a **pending** fabric not yet committed by CommissioningComplete (e.g. after an AddNOC, in the context of the new fabric index)

Then, basically, the CommissioningARL ought to block all aspects of the Node which are not accessible unless there would be an override. This can be known a-priori.

Reply:
This should be resolved now.